frontend/src/pages/EditProfile.js

```diff
@@ -0,0 +1,206 @@
+import React, { useState, useEffect } from "react";
+
+import { blue } from "@ant-design/colors";
+import {
+  Row,
+  Col,
+  Card,
+  Avatar,
+  Form,
+  Input,
+  Empty,
+  Button,
+  Upload,
+  message,
+} from "antd";
+import { UploadOutlined } from "@ant-design/icons";
+import { useSelector, useDispatch } from "react-redux";
+import { useNavigate } from "react-router-dom";
+
+import { authenticate } from "../helpers/auth";
+import { editUserProfile } from "../helpers/api";
+import { checkFileUpload, isValidFile } from "../helpers/imageHelper";
+import { setCurrentUser } from "../store/currentUserSlice";
+import HomeLayout from "../layouts/HomeLayout";
+
+const EditProfile = () => {
+  const { user } = useSelector((state) => state.currentUser);
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+
+  const [avatarUrl, setAvatarUrl] = useState(undefined);
+  const [selectedImg, setSelectedImg] = useState("");
+
+  useEffect(() => {
+    authenticate(navigate, dispatch);
+    if (!user.id) return;
+
+    setSelectedImg(user.avatar_url);
+  }, [dispatch, navigate, user.id, user.avatar_url]);
+
+  const uploadProps = {
```

Comment:
- move out _local states, useEffects, custom functions_ to a hook file (research and study "reactjs custom hooks")